model/inline/src/main/scala/aqua/model/inline/state/Exports.scala

```diff
@@ -59,18 +124,91 @@ trait Exports[S] extends Scoped[S] {
 object Exports {
   def apply[S](implicit exports: Exports[S]): Exports[S] = exports
 
+  // Get last linked VarModel
+  def getLastHelper[T](name: String, state: Map[String, ValueModel], ex: VarModel => T): Option[T] = {
+    state.get(name) match {
+      case Some(vm@VarModel(n, _, _)) =>
+        if (name == n) Option(ex(vm))
+        else getLastHelper(n, state, ex).orElse(Option(ex(vm)))
+      case n =>
+        None
+    }
+  }
+
   object Simple extends Exports[Map[String, ValueModel]] {
 
-    // Exports[Map[NonEmptyList[String], ValueModel]]
+    // Make links from one set of abilities to another (for ability assignment)
+    private def getAbilityPairs(oldName: String, newName: String, at: AbilityType, state: Map[String, ValueModel]): NonEmptyList[(String, VarModel)] = {
+      at.fields.toNel.flatMap {
+        case (n, at@AbilityType(_, _)) =>
+          val newFullName = AbilityType.fullName(newName, n)
+          val oldFullName = AbilityType.fullName(oldName, n)
+          getAbilityPairs(oldFullName, newFullName, at, state)
+        case (n, t) =>
+          val newFullName = AbilityType.fullName(newName, n)
+          val oldFullName = AbilityType.fullName(oldName, n)
+          // put link on last variable in chain
+          val lastVar = Exports.getLastHelper(oldFullName, state, identity)
+          NonEmptyList.of((newFullName, lastVar.getOrElse(VarModel(oldFullName, t))))
+      }
+    }
 
     override def resolved(
       exportName: String,
       value: ValueModel
-    ): State[Map[String, ValueModel], Unit] = State.modify(_ + (exportName -> value))
+    ): State[Map[String, ValueModel], Unit] = State.modify { state =>
+      value match {
+        case vm@Ability(name, at, property) if property.isEmpty =>
+          val pairs = getAbilityPairs(name, exportName, at, state)
+          state ++ pairs.toList.toMap
+        case _ => state + (exportName -> value)
+      }
+    }
+
+    override def getLast(name: String): State[Map[String, ValueModel], Option[String]] =
+      State.get.map(st => getLastHelper(name, st, _.name))
 
     override def resolved(exports: Map[String, ValueModel]): State[Map[String, ValueModel], Unit] =
```

Comment:
This is not in sync with single `resolved`. I think it is better to implement one in terms of another?